kibana/mapping.py

```diff
@@ -27,10 +27,11 @@ def iteritems(d):
 
 
 class KibanaMapping():
-    def __init__(self, index, index_pattern, host, debug=False):
+    def __init__(self, index, index_pattern, host, map_ua=None, debug=False):
         self.index = index
         self._index_pattern = index_pattern
         self._host = host
+        self.map_ua = map_ua,
```

Comment:
Let's call the setting/conversion function here once, to avoid relying on isinstance of str and leverage a single session instead of one session per request. Also, then instead of self.get_map_ua() we can call the attribute directly.